src/wallet/db.h

```diff
@@ -143,9 +143,12 @@ class BerkeleyDatabase
         return MakeUnique<BerkeleyDatabase>(std::make_shared<BerkeleyEnvironment>(), "");
     }
 
-    /** Open the database if it is not already opened */
+    /** Open the database if it is not already opened. Increments mapFileUseCount */
     void Open(const char* mode);
 
+    /** Indicate that database user has stopped using the database. Decrement mapFileUseCount */
```

Comment:
I think you are confusing `BerkeleyBatch` and `BerkeleyDatabase`. `Release` is called by `BerkeleyBatch::Close`, not `BerkeleyDatabase::Close`.

In a later commit, a corresponding `Acquire` function is added. In this commit, the thing that `Acquire` does is part of `Open`. The idea with the refcount is thusly: when something tries to use the database, it increments the refcount. When it is done, it decrements the refcount. The goal is to prevent the database from closing while something is still using the database. When `WalletBatch` is initialized, it calls `BerkeleyDatabase::Open` via `BerkeleyBatch`'s constructor. When `WalletBatch` is done and no longer being used, it calls `BerkeleyBatch::Close` which calls `BerkeleyDatabase::Release`. Then both the `BerkeleyBatch` and `WalletBatch` are destructed. The `BerkeleyDatabase` remains open so that future uses of the database don't need to reopen it and for other concurrent `WalletBatch`s to do their work.

So the flow with `BerkeleyDatabase` is `Open`, changes, `Release`, `Close`.

Reply:
Done